client.go

```diff
@@ -50,43 +46,61 @@ func (c *Client) Get(ctx context.Context, req *Request) (*GetResult, error) {
 		req.Mode = ModeAny
 	}
 
-	var err error
-	req.Src, err = Detect(req.Src, req.Pwd, c.Detectors)
-	if err != nil {
-		return nil, err
-	}
-
-	var force string
-	// Determine if we have a forced protocol, i.e. "git::http://..."
-	force, req.Src = getForcedGetter(req.Src)
-
 	// If there is a subdir component, then we download the root separately
 	// and then copy over the proper subdir.
-	var realDst, subDir string
-	req.Src, subDir = SourceDirSubdir(req.Src)
-	if subDir != "" {
+	req.Src, req.subDir = SourceDirSubdir(req.Src)
+	if req.subDir != "" {
 		td, tdcloser, err := safetemp.Dir("", "getter")
 		if err != nil {
 			return nil, err
 		}
 		defer tdcloser.Close()
 
-		realDst = req.Dst
+		req.realDst = req.Dst
 		req.Dst = td
 	}
 
-	req.u, err = urlhelper.Parse(req.Src)
-	if err != nil {
-		return nil, err
+	var multierr []error
+	for _, g := range c.Getters {
+		ok, err := Detect(req, g)
+		if err != nil {
+			return nil, err
+		}
+		if !ok {
+			continue
```

Comment:
What do you think about: 
```suggestion
		if !ok {
			// the request should not be processed by that getter
			continue
```